jujugui/static/gui/src/app/components/account/account.js

```diff
@@ -25,10 +25,32 @@ YUI.add('account', function() {
     propTypes: {
       acl: React.PropTypes.object.isRequired,
       addNotification: React.PropTypes.func.isRequired,
-      getUser: React.PropTypes.func.isRequired,
+      getCloudCredentialNames: React.PropTypes.func.isRequired,
+      getCloudProviderDetails: React.PropTypes.func.isRequired,
+      getUser: React.PropTypes.func,
+      listClouds: React.PropTypes.func.isRequired,
+      revokeCloudCredential: React.PropTypes.func.isRequired,
+      showPay: React.PropTypes.bool,
+      user: React.PropTypes.string.isRequired,
       userInfo: React.PropTypes.object.isRequired
     },
 
+    /**
+      Generate the payment details section.
+
+      @method _generatePaymentDetails
+    */
+    _generatePaymentDetails: function() {
+      if (this.props.showPay) {
+        return (
+          <juju.components.AccountPaymentMethod
+            acl={this.props.acl}
+            addNotification={this.props.addNotification}
+            getUser={this.props.getUser}
+            username={this.props.userInfo.profile} />);
+      }
```

Comment:
This should return `null` as we had discussed - unless that causes issues with it rendering "null"